sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2185,7 +2185,10 @@ class Analyzer(override val catalogManager: CatalogManager)
               externalFunctionNameSet.add(fullName)
               f
             } else {
-              throw QueryCompilationErrors.noSuchFunctionError(nameParts, f, Some(fullName))
+              throw QueryCompilationErrors.unresolvedRoutineError(
+                nameParts,
+                Seq("system.builtin", "system.session", catalog.name()),
```

Comment:
Shouldn't we backtick these?
Also shouldn't the last one include the current schema/database?

Reply:
+1

Reply:
> Shouldn't we backtick these?

I added backticks.

> Also shouldn't the last one include the current schema/database?

done. Added to the current namespace.